tests/src/test/java/alluxio/client/cli/job/DistributedCommandsStatsTest.java

```diff
@@ -56,6 +56,71 @@
                 .setExistsMs(SLEEP_MS)
                 .setMkdirsMs(SLEEP_MS)));
 
+  @Test
```

Comment:
this is a single case functional test that ensures the metrics output of the load job is what we expect. it does not test against one of the issues that were fixed here, where all job types went through the full complete status codepath of calling `fileSystem.getStatus()` even though only the 3 relevant jobs were returning a non-null path to getStatus on. however do let me know if i don't understand the problem correctly; this is what i guessed from reading your fix...

when i suggested to write a unit test, my motivation was "how do we make sure we don't encounter the same problem again" and this unit test doesn't solve this question. the test i was proposing has the psuedocode:
```
mock the FileSystem fileSystem argument in public static void incrementForCompleteStatusWithRetry()
for every job type
  submit job
  if job is of type Load, Persist, or Migrate
    check that the mocked fileSystem.getStatus is called
  else
    check that the mocked fileSystem.getStatus is not called
```
it may take more refactoring to be able to write this unit test, but i am not familiar enough with the codebase to estimate the amount of effort. it would be very sad if it is not possible and these are the types of code tech debts we should invest more time in rather than declaring the problem to be too hard.

Reply:
@Xenorith Thanks very much for the comment! I see your point, just let me clarify something quickly: In the last commit in the PR, I added `Failure counter` incrementation in the PlanCoordinator class. The fix is to remove the loop iteration for `Failure counter` where the code did repeatedly instance type check (see  `config.getJobConfigs().forEach(jobConfig -> { incrementForFailStatus(jobType); })`, and now directly add the count values to the counter, so we get rid of the repeated job type checks. This loop will slow down the code run speed significantly when the job is a batch job and the batch size is very big. 
For the unit test, it shows the counter values  of different job types after the job finishes. In our case, we tested on a `LoadConfig`, and if its Complete status count is 1, it means only the code block 
`if (config instanceof LoadConfig) {
      filePath = ((LoadConfig) config).getFilePath();
      JOB_DISTRIBUTED_LOAD_SUCCESS.inc(DEFAULT_INCREMENT_VALUE);
      JOB_DISTRIBUTED_LOAD_FILE_COUNT.inc(DEFAULT_INCREMENT_VALUE);
      fileSize = getFileSize(filePath, fileSystem, retryPolicy);
      JOB_DISTRIBUTED_LOAD_FILE_SIZE.inc(fileSize);
      JOB_DISTRIBUTED_LOAD_RATE.mark(fileSize);
    } ` 
is called. If not, other counters like MIGRATE_JOB_SUCCESS and ASYNC_PERSIST_SUCCESS etc, will have non-zero values. From the counter value comparison, we can see only DistributedLoad counters have non-zero values, so that means only the code of `fileSystem.getStatus() ` inside `getFileSize` in the LoadConfig case (the if-else statement) was called.
That test is somewhat an indirect way to test the correctness, by comparing the counter values of different job types. And I'll see how to create another test to mock your proposed test.

Reply:
I think JobShellTest is more a integration test so it's kind of hard too test `funcion not called`. Probably just do this in a single unit test for this class and inject mock file system, which would be easier.

Reply:
Yeah, let me try this way, thanks!